key/src/main/java/software/amazon/kms/key/UpdateHandler.java

```diff
@@ -58,9 +58,12 @@ public UpdateHandler(final ClientBuilder clientBuilder,
                 .updateKeyDescription(proxy, proxyClient, previousModel, model, callbackContext))
             .then(progress -> keyHandlerHelper
                 .updateKeyPolicy(proxy, proxyClient, previousModel, model, callbackContext))
-            .then(progress -> softFailAccessDenied(() -> keyHandlerHelper
-                .updateKeyTags(proxy, proxyClient, model, tags, callbackContext),
-                model, callbackContext))
+            .then(progress -> {
+                if (!tags.equals(request.getPreviousResourceTags())) {
+                    keyHandlerHelper.updateKeyTags(proxy, proxyClient, model, tags, callbackContext);
+                }
+                return ProgressEvent.progress(model, callbackContext);
```

Comment:
May be my ignorance but what is the intent of this return ? Seems like blocks are not doing a return.